src/fp3.rs

```diff
@@ -75,11 +75,11 @@ impl fmt::Debug for Fp3 {
             .collect::<Vec<String>>()
             .join(" + ");
 
-        if elem_rep.is_empty() {
-            write!(f, "0")?; // Handle the case where all coefficients are zero
-        } else {
-            write!(f, "{}", elem_rep)?;
-        }
+        if *self == Fp3::zero() {
+            return write!(f, "0"); // Handle the case where all coefficients are zero
+        } 
```

Comment:
Ah sorry, maybe I wasn't being clear, but I was suggesting to have it as an initial check, at the start of the function (it is not necessary to process the list of coefficients if the element is zero).

Reply:
Makes sense. I should have done that :)